scripts/lint_naming_conventions.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+
+# Script used to review jenkins jobs and report job names
+# not following conventions
+# https://github.com/rcbops/rpc-gating#naming-conventions
+
+import argparse
+import os
+import re
+import sys
+import yaml
+
+
+def parse_args():
+    description = "Parses JJB files to ensure naming conventions followed"
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument("--filename", required=True, help="File to parse")
+    parser.add_argument("--parse-jjb", required=False,
+                        help="Parse jjb file for names",
+                        action='store_true')
+    return parser.parse_args()
+
+
+def parse_jjb_file(filename):
+    file_yaml = yaml.safe_load(open(filename).read())
+    for item in file_yaml:
+        if "project" in item:
+            # check project name
+            parse_job_name(item["project"]["name"], filename)
+            # check project jobs names
+            for job in item["project"]["jobs"]:
+                parse_job_name(job, filename)
+
+        if "job-template" in item:
+            # check job-template name
+            parse_job_name(item["job-template"]["name"], filename)
+
+        if "job" in item:
+            # check individual job
+            parse_job_name(item["job"]["name"], filename)
+
+
+def parse_job_name(job_name, file_name):
+    regex = '^([a-zA-Z0-9]+-?)+([_]([{][a-zA-Z0-9]+[}]-?)+)?$'
+    match = re.match(regex, job_name)
+    if not match:
+        sys.stderr.write("Lint error on %s in %s\n" % (job_name, file_name))
```

Comment:
It would be helpful to know that its the job name that failed a check, rather than the job itself. Maybe:
```
"Lint error: Job name {job_name} from {file_name} doesn't confirm to rpc-gating naming standards".format(job_name=job_name, file_name=file_name)
```